pkg/api/apikey.go

```diff
@@ -114,59 +113,15 @@ func (hs *HTTPServer) DeleteAPIKey(c *contextmodel.ReqContext) response.Response
 // see: https://grafana.com/docs/grafana/next/administration/api-keys/#migrate-api-keys-to-grafana-service-accounts-using-the-api.
 //
 // Responses:
-// 200: postAPIkeyResponse
-// 400: badRequestError
-// 401: unauthorisedError
-// 403: forbiddenError
-// 409: conflictError
-// 500: internalServerError
+// 301: statusMovedPermanently
 func (hs *HTTPServer) AddAPIKey(c *contextmodel.ReqContext) response.Response {
-	cmd := apikey.AddCommand{}
-	if err := web.Bind(c.Req, &cmd); err != nil {
-		return response.Error(http.StatusBadRequest, "bad request data", err)
-	}
-	if !cmd.Role.IsValid() {
-		return response.Error(http.StatusBadRequest, "Invalid role specified", nil)
-	}
-	if !c.SignedInUser.GetOrgRole().Includes(cmd.Role) {
-		return response.Error(http.StatusForbidden, "Cannot assign a role higher than user's role", nil)
-	}
+	// Set the Location header to the new URL
+	hs.log.Warn("Deprecated API endpoint called", "path", c.Req.URL.Path)
```

Comment:
nit: maybe something like 
```suggestion
	hs.log.Warn("Removed legacy API key endpoint called", "path", c.Req.URL.Path)
```

Reply:
i like the use of the word "legacy". Does legacy mean that they are in use to you? I am asking freely here,

could work with words from my friend AI:

Terms and Phrases:

	1.	Deprecated: Indicates that the feature is outdated and should not be used, as it will be removed in the future.
	•	Example: “This API key is deprecated and will be removed in a future release.”
	2.	Legacy: Refers to older features that are still available but are no longer recommended or maintained.
	•	Example: “You are using a legacy API key that is no longer supported.”
	3.	Obsolete: Suggests that the feature is no longer in use and has been replaced by something newer.
	•	Example: “This functionality is obsolete and should not be used in new implementations.”
	4.	Removed: Indicates that the feature is no longer available in the system.
	•	Example: “The API key has been removed and is no longer valid.”
	5.	End of Life (EOL): Used to signify that the feature has reached the end of its lifecycle and is no longer supported.
	•	Example: “API keys have reached end of life and are no longer available.”
	6.	No Longer Supported: Indicates that the feature is not maintained or supported anymore.
	•	Example: “API keys are no longer supported; please migrate to OAuth.”
	7.	Phased Out: Refers to the gradual removal of a feature.
	•	Example: “API keys are being phased out; please transition to the new authentication method.”
	8.	Superseded: Suggests that the feature has been replaced by a newer one.
	•	Example: “API keys have been superseded by token-based authentication.”
	
	
From this i like "Removed, Superseded, No longer available"